src/client.rs

```diff
@@ -208,6 +207,10 @@ impl DatagoClient {
         }
 
         // Try to fetch a new sample from the queue
+        // The client will timeout if zero sample is received in 5 minutes
+        // At this point it will stop and wrap everything up
+        const TIMEOUT: std::time::Duration = std::time::Duration::from_secs(300);
```

Comment:
this is the only change really, but as discussed I think the root of the issue was not there but in many clients unknowingly competing for the machine resources